python/tskit/ms.py

```diff
@@ -0,0 +1,114 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Write the genotypes from tree sequences to ms-style output
+"""
+
+
+class msWriter:
+    """
+    Writes a ms representation of the genotypes in tree sequence to a
+    file-like object.
+    """
+
+    def __init__(
+        self,
+        tree_sequence,
+        print_trees=False,
+        precision=4,
+        mutation_rate=0,
+        num_replicates=1,
+        write_header=True,
+    ):
+        self._tree_sequence = tree_sequence
+        self._num_replicates = num_replicates
+        self._mutation_rate = mutation_rate
+        self._precision = precision
+        self._print_trees = print_trees
+        self._write_header = write_header
+
+    def print_trees(self, output):
+        """
+        Print out the trees in ms-format from the specified tree sequence.
+        When 'invisible' recombinations occur ms prints out copies of the
+        same tree. Therefore, we must keep track of all breakpoints from the
+        simulation and write out a tree for each one.
+        """
+        tree = next(self._tree_sequence.trees())
+        newick = tree.newick(precision=self._precision)
+        print(f"[{tree.span}]", newick, file=output)
+
+    def __write_header(self, output):
+        print(
+            "ms {} {}".format(
+                self._tree_sequence.sample_size, max(self._num_replicates, 1)
+            ),
+            file=output,
+        )
+        print("{}".format(999), file=output)
+
+    def write(self, output):
+
+        if self._write_header is True:
+            self.__write_header(output)
+
+        tree_sequence = self._tree_sequence
+        print(file=output)
+        print("//", file=output)
+        if self._print_trees is True:
+            self.print_trees(output)
+        if self._mutation_rate > 0:
+            s = tree_sequence.get_num_sites()
+            print("segsites:", s, file=output)
+            if s != 0:
+                print("positions: ", end="", file=output)
+                positions = [
+                    variant.position / (tree_sequence.sequence_length)
+                    for variant in tree_sequence.variants()
+                ]
+                for position in positions:
+                    print(
+                        "{0:.{1}f}".format(position, self._precision),
+                        end=" ",
+                        file=output,
+                    )
+                print(file=output)
+
+                genotypes = tree_sequence.genotype_matrix()
+                for k in range(tree_sequence.num_samples):
+                    tmp_str = "".join(map(str, genotypes[:, k]))
+                    if set(tmp_str).issubset({"0", "1", "-"}):
+                        print(tmp_str, file=output)
+                    #################################################
+                    # Introducing an error to exit if the sequence  #
```

Comment:
Comment isn't needed now we have the exception text.

Reply:
Removed the comment.